msal/tests/Test.MSAL.NET.Unit/AuthorityAliasesTests.cs

```diff
@@ -0,0 +1,216 @@
+                Method = HttpMethod.Post,
+                ResponseMessage = MockHelpers.CreateSuccessTokenResponseMessage()
+            });
+
+            AuthenticationResult result = app.AcquireTokenAsync(TestConstants.Scope).Result;
+
+            // make sure that all cache entities are stored with "preferred_cache" environment
+            // (it is taken from metadata in instance discovery response)
+            ValidateCacheEntitiesEnvironment(app.UserTokenCache, TestConstants.ProdPrefCacheEnv);
+
+            // silent request targeting at, should return at from cache for any environment alias
+            foreach (var envAlias in TestConstants.ProdEnvAliases)
+            {
+                result = app.AcquireTokenSilentAsync(TestConstants.Scope,
+                    app.GetAccountsAsync().Result.First(),
+                    string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", envAlias, TestConstants.Utid),
+                    false).Result;
+
+                Assert.IsNotNull(result);
+            }
+
+            // mock for openId config request for tenant spesific authority
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Url = string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/v2.0/.well-known/openid-configuration",
+                    TestConstants.ProdPrefNetworkEnv, TestConstants.Utid),
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityUtidTenant)
+            });
+
+            // silent request targeting rt should find rt in cahce for authority with any environment alias
+            foreach (var envAlias in TestConstants.ProdEnvAliases)
+            {
+                HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+                {
+                    Url = string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/oauth2/v2.0/token",
+                        TestConstants.ProdPrefNetworkEnv, TestConstants.Utid),
+                    Method = HttpMethod.Post,
+                    PostData = new Dictionary<string, string>()
+                    {
+                        {"grant_type", "refresh_token"}
+                    },
+                    // return not retriable status code
+                    ResponseMessage = MockHelpers.CreateInvalidGrantTokenResponseMessage()
+                });
+
+                try
+                {
+                    result = app.AcquireTokenSilentAsync(TestConstants.ScopeForAnotherResource,
+                        app.GetAccountsAsync().Result.First(),
+                        string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", envAlias, TestConstants.Utid),
+                        false).Result;
+                }
+                catch (AggregateException ex)
+                {
+                    Assert.IsNotNull(ex.InnerException);
```

Comment:
The intent of the test is not clear - do we expect a UIRequiredException or do we expect the result to be retured? Please refactor so that the requests that result in an exception fail if a result is returned and vice versa.

Reply:
good catch, will change